packages/core/test/result.test.ts

```diff
@@ -1381,6 +1454,30 @@ describe('Result', () => {
       })
     })
 
+    describe('.isOpen()', () => {
+      it('should be true', async () => {
+        expect(result.isOpen()).toBe(true)
+
+        // consume the stream to avoid unhaddled errors
+        try {
+          await result.summary()
+        } catch (error) {
+        }
+      })
+
+      it('should be false after any interactio with the stream', async () => {
+        const it = result[Symbol.asyncIterator]()
+        
+        try {
+          await it.next()
+        } catch (error) {
+          // this's fine
```

Comment:
When/why can it fail?

Reply:
This is iterating over a failed result, so it will always thrown and this is fine.